ruggedpod_api/service_auth.py

```diff
@@ -47,18 +51,31 @@ class Cypher(object):
     __metaclass__ = Singleton
 
     def __init__(self):
-        self.key = Fernet.generate_key()
-        self.cipher_suite = Fernet(self.key)
+        self.key = hashlib.sha256(auth['secret_key'].encode()).digest()
 
-    def encrypt(self, text):
-        return self.cipher_suite.encrypt(b"%s" % text)
+    def encrypt(self, raw):
+        iv = Random.new().read(AES.block_size)
+        cipher = AES.new(self.key, AES.MODE_CBC, iv)
+        return base64.b64encode(iv + cipher.encrypt(self._pad(raw)))
 
-    def decrypt(self, cipher):
+    def decrypt(self, enc):
         try:
-            return self.cipher_suite.decrypt(cipher.encode("ascii"))
+            enc = base64.b64decode(enc)
```

Comment:
I would not use the parameter to represent both encrypted base64 encoded string and encrypted base64 decoded string.
Maybe you should use two variables "encoded" and "encrypted" ?